logstash.go

```diff
@@ -8,22 +8,36 @@ import (
 
 // Hook represents a connection to a Logstash instance
 type Hook struct {
-	conn    net.Conn
-	appName string
+	conn             net.Conn
+	appName          string
+	alwaysSentFields logrus.Fields
 }
 
 // NewHook creates a new hook to a Logstash instance, which listens on
 // `protocol`://`address`.
 func NewHook(protocol, address, appName string) (*Hook, error) {
+	return NewHookWithFields(protocol, address, appName, make(logrus.Fields))
+}
+
+// NewHookWithFields creates a new hook to a Logstash instance, which listens on
+// `protocol`://`address`. alwaysSentFields will be sent with every log entry.
+func NewHookWithFields(protocol, address, appName string, alwaysSentFields logrus.Fields) (*Hook, error) {
 	conn, err := net.Dial(protocol, address)
 	if err != nil {
 		return nil, err
 	}
-	return &Hook{conn: conn, appName: appName}, nil
+	return &Hook{conn: conn, appName: appName, alwaysSentFields: alwaysSentFields}, nil
 }
 
 func (h *Hook) Fire(entry *logrus.Entry) error {
 	formatter := LogstashFormatter{Type: h.appName}
+
+	// Add in the alwaysSentFields. We don't override fields that are already set.
+	for k, v := range h.alwaysSentFields {
+		if _, inMap := entry.Data[k]; !inMap {
+			entry.Data[k] = v
```

Comment:
`entry`'s are released after every log so I don't think it's a problem to be mutating `.Data` here